game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/AggregateResults.java

```diff
@@ -107,103 +195,362 @@ public double getAverageTuvSwing(
+                                result.getRemainingDefendingUnits(), defenderCostsForTuv))
+                .toArray());
+    // If there are no battles, stdDev.getResult() returns Double.NaN.
+    return Double.isNaN(m) ? 0 : defenderStartingTuv - attackerStartingTuv + m;
   }
 
-  double getAverageDefendingUnitsLeft() {
-    if (results.isEmpty()) {
-      return 0.0;
-    }
-    return results.stream()
-            .map(BattleResults::getRemainingDefendingUnits)
-            .mapToDouble(Collection::size)
-            .sum()
-        / results.size();
+  /**
+   * Returns the standard deviation of the TUV swing.
+   *
+   * <p>Note: The result of this function only makes sense for battle with the same initial setup,
+   * e.g. results coming from simulations of the same battle. The {@code attackers} and {@code
+   * defenders} are the units attacking and defending in this setup, respectively.
+   *
+   * <p>{@code attacker}, {@code defender} and {@code data} are used to derive the TUV values of the
+   * units.
+   *
+   * @param attacker the attacking player
+   * @param defender the defending player
+   * @param data the game data
+   * @return the standard deviation for the TUV swing of the battles.
+   */
+  public double getAverageTuvSwingStandardDeviation(
+      final GamePlayer attacker, final GamePlayer defender, final GameData data) {
+    // The variance is invariant under a constant shift.  Thus, the variance of
+    // the TUV of left-over units and the variance of the TUV of lost units is
+    // the same as they differ only be the (constant) total TUV the units at
+    // the start of the battle.
+    final IntegerMap<UnitType> attackerCostsForTuv = TuvUtils.getCostsForTuv(attacker, data);
+    final IntegerMap<UnitType> defenderCostsForTuv = TuvUtils.getCostsForTuv(defender, data);
+    // The 'true' argument means calculate the variance using bias correction.
+    final StandardDeviation stdDev = new StandardDeviation(true);
+    final double s =
+        stdDev.evaluate(
+            results.stream()
+                .mapToDouble(
+                    result ->
+                        TuvUtils.getTuv(result.getRemainingAttackingUnits(), attackerCostsForTuv)
+                            - TuvUtils.getTuv(
+                                result.getRemainingDefendingUnits(), defenderCostsForTuv))
+                .toArray());
+    return Double.isNaN(s) ? 0 : s;
   }
 
-  double getAverageDefendingUnitsLeftWhenDefenderWon() {
-    if (results.isEmpty()) {
-      return 0.0;
-    }
-    double count = 0;
-    double total = 0;
-    for (final BattleResults result : results) {
-      if (result.defenderWon()) {
-        count += result.getRemainingDefendingUnits().size();
-        total += 1;
-      }
-    }
-    if (total <= 0) {
-      return 0;
-    }
-    return count / total;
+  /**
+   * Returns the average number of attacking units surviving the battles.
+   *
+   * @return the average of surviving attacking units
```

Comment:
nit, if the main paragraph of a javadoc repeats the `@return` part, typically I'll leave the `@return` off. A tell-tale sign is when the comment begins with the word 'returns'. We do not generate javadocs in this project and it's super rare for anyone to view the rendered javadocs in any context, hence the target audience is really another Dev, in an IDE reading this with the code right below it.